sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_message_backcompat.py

```diff
@@ -0,0 +1,238 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+# pylint: disable=too-many-lines
+from enum import Enum
+
+from ._encode import encode_payload
+from .utils import get_message_encoded_size
+from .error import AMQPError
+from .message import Message, Header, Properties, BatchMessage
+#from uamqp import constants, errors
+
+
+class MessageState(Enum):
+    WaitingToBeSent = 0
+    WaitingForSendAck = 1
+    SendComplete = 2
+    SendFailed = 3
+    ReceivedUnsettled = 4
+    ReceivedSettled = 5
+
+    def __eq__(self, __o: object) -> bool:
+        try:
+            return self.value == __o.value
+        except AttributeError:
+            return super().__eq__(__o)
+
+
+class MessageAlreadySettled(Exception):
+    pass
+
+
+DONE_STATES = (MessageState.SendComplete, MessageState.SendFailed)
+RECEIVE_STATES = (MessageState.ReceivedSettled, MessageState.ReceivedUnsettled)
+PENDING_STATES = (MessageState.WaitingForSendAck, MessageState.WaitingToBeSent)
+
+
+class LegacyMessage(object):
+    def __init__(self, message, **kwargs):
+        self._message = message
+        self.state = MessageState.SendComplete
+        self.idle_time = 0
+        self.retries = 0
+        self._settler = kwargs.get('settler')
+        self._encoding = kwargs.get('encoding')
+        self.delivery_no = kwargs.get('delivery_no')
+        self.delivery_tag = kwargs.get('delivery_tag') or None
+        self.on_send_complete = None
+        self.properties = LegacyMessageProperties(self._message.properties)
+        self.application_properties = self._message.application_properties
+        self.annotations = self._message.annotations
+        self.header = LegacyMessageHeader(self._message.header)
+        self.footer = self._message.footer
```

Comment:
for properties and header, these result in an error if self._message.header/properties is None.
```suggestion
        self.properties = LegacyMessageProperties(self._message.properties) if self._message.properties else None
        self.header = LegacyMessageHeader(self._message.header) if self._message.header else None
```
for the other ones, wondering if they should be like in AmqpAnnotatedMessage, like this? or if they should just be None, as they are here?
```
self.application_properties = self._message.application_properties if self._message.application_properties else {}
self.annotations = self._message.annotations if self._message.annotations else {}
self.footer = self._message.footer if self._message.footer else {}
self.delivery_annotations = self._message.delivery_annotations if self._message.delivery_annotations else {}
```